src/bp-templates/bp-nouveau/js/buddypress-media.js

```diff
@@ -375,6 +403,797 @@ window.bp = window.bp || {};
 				$( '#bp-media-edit-child-folder' ).show();
 			}
 
+			// Set up the draft keys/intervals/display data when Buddyboss theme is enabled.
+			if ( this.is_bb_theme ) {
+				$( document ).on( 'bbp_after_load_topic_form', this.setupOnOpenTopicReplyModal.bind( this ) );
+				$( document ).on( 'bbp_after_load_reply_form', this.setupOnOpenTopicReplyModal.bind( this ) );
+				$( document ).on( 'bbp_after_load_inline_reply_form', this.setupOnOpenTopicReplyModal.bind( this ) );
+				$( document ).on( 'bbp_after_close_topic_reply_form', this.clearOnCloseTopicReplyModal.bind( this ) );
+			} else {
+				// Set up the intervals.
+				$( window ).on( 'load', this.setupTopicReplyDraftIntervals.bind( this ) );
+				bp.Nouveau.Media.displayTopicReplyDraft();
+			}
+
+			if ( this.bbp_is_reply_edit || this.bbp_is_topic_edit || this.bbp_is_forum_edit ) {
+				bp.Nouveau.Media.topic_reply_draft.allow_delete_media = true;
+			}
```

Comment:
move all those code and related code to the file `src/bp-forums/templates/default/js/editor.js`
OR create a new one as if the media component disabled this file was not called.